packages/p2p-v2/src/components/FileDropzone/FileDropzone.tsx

```diff
@@ -0,0 +1,127 @@
+import { Text } from '@deriv-com/ui';
+import { FadeInMessage } from './FadeInMessage';
+import { PreviewSingle } from './PreviewSingle';
+import './FileDropzone.scss';
+
+// TODO: Remove this and other associated files once FileDropzone component from deriv-com/ui is completed
+const FileDropzone = ({ className, noClick = false, ...props }: TFileDropzone) => {
+    const {
+        accept,
+        errorMessage,
+        filenameLimit,
+        hoverMessage,
+        maxSize,
+        message,
+        multiple,
+        onDropAccepted,
+        onDropRejected,
+        validationErrorMessage,
+        value,
+    } = props;
+
+    const RenderErrorMessage = useCallback(
+        ({ open }: DropzoneRef) => {
+            if (noClick && typeof message === 'function') return <>{message(open)}</>;
+
+            return <>{message}</>;
+        },
+        [message, noClick]
+    );
+
+    const RenderValidationErrorMessage = useCallback(
+        ({ open }: DropzoneRef) => {
+            if (typeof validationErrorMessage === 'function') return <>{validationErrorMessage(open)}</>;
+
+            return <>{validationErrorMessage}</>;
+        },
+        [validationErrorMessage]
+    );
+
+    const dropzoneRef = useRef(null);
+
+    return (
+        <Dropzone
+            // accept prop is same as native HTML5 input accept - e.g - 'image/png'
+            accept={accept}
+            // set maximum size limit for file, in bytes (binary)
+            maxSize={maxSize}
+            // allow multiple uploads
+            multiple={multiple || false}
+            noClick={noClick}
+            // sends back accepted files array
+            onDropAccepted={onDropAccepted}
+            // sends back rejected files array
+            onDropRejected={onDropRejected}
+        >
+            {({ getInputProps, getRootProps, isDragAccept, isDragActive, isDragReject, open }) => (
+                <div
+                    {...getRootProps()}
+                    className={classNames('p2p-v2-file-dropzone', className, {
+                        'p2p-v2-file-dropzone--has-error': (isDragReject || !!validationErrorMessage) && !isDragAccept,
+                        'p2p-v2-file-dropzone--has-file': isDragActive || value.length > 0,
+                        'p2p-v2-file-dropzone--is-active': isDragActive,
+                        'p2p-v2-file-dropzone--is-noclick': noClick,
+                    })}
+                    ref={dropzoneRef}
+                >
+                    <input {...getInputProps()} data-testid='dt_p2p_v2_file_upload_input' />
+                    <div className='p2p-v2-file-dropzone__content'>
+                        <FadeInMessage
+                            // default message when not on hover or onDrag
+                            isVisible={!isDragActive && !!message && value.length < 1 && !validationErrorMessage}
+                            noText={noClick}
+                            timeout={150}
```

Comment:
should extract this into a separate constant:
```suggestion
                            timeout={DROPZONE_TIMEOUT}
```